src/sardana/macroserver/msmacromanager.py

```diff
@@ -1290,9 +1294,31 @@ def __runXMLMacro(self, xml):
     _runXMLMacro = __runXMLMacro
 
     def runMacro(self, macro_obj):
+
         name = macro_obj._getName()
         desc = macro_obj._getDescription()
         door = self.door
+        self.macro_obj = macro_obj
+
+        if macro_obj.getParentMacro() is None:
+            try:
+                self.logging_onoff = macro_obj.getEnv("LogMacroOnOff")
+            except:
+                self.logging_onoff = False
+            if self.logging_onoff:
+                try:
+                    self.logging_path = macro_obj.getEnv("LogMacroPath")
+                except:
+                    macro_obj.setEnv("LogMacroPath", "/tmp")
+                    self.logging_path = macro_obj.getEnv("LogMacroPath")
+                log_file = self.logging_path + "/spock_session.log"
+                self.macro_obj.fileHandler = logging.FileHandler(log_file)
```

Comment:
I think that we don't need these *monkey patching* (`fileHandler` and `logger`). Actually it would be enough to keep the `fileHandler` reference locally if we remove the handler in `runMacro` as well (see my previous comment).

Reply:
In fact, this is there because it was needed in my previous implementation (without using the Handler) and I did not clean the files ... waiting for what would be decided to be used. I can remove it now.

Reply:
I think you can go ahead with this as well.

Reply:
I'm thinking we are getting close to the integration of this PR, could you already remove these references? This will make the review process cleaner.

Reply:
I thinks it is done.

Reply:
I'm not sure if this assignments are necessary:
`macro_obj.fileHandler = logging.handlers.RotatingFileHandler(log_file, backupCount = bck_counts)`
`macro_obj.logger = macro_obj.getLogger()`.
Or in other words does the macro_obj require special references to `fileHandler` and `logger`.

Reply:
I remove them, they are not needed.

Reply:
Great thanks!